cassandra-management/src/main/java/org/hawkular/cassandra/management/CassandraMaintenanceJob.java

```diff
@@ -0,0 +1,130 @@
+ * events in the Cassandra cluster, and the job looks at that log to determine what if any maintenance needs to be
+ * performed.
+ *
+ * @author jsanda
+ */
+public class CassandraMaintenanceJob implements Func1<JobDetails, Completable> {
+
+    private static Logger logger = Logger.getLogger(CassandraMaintenanceJob.class);
+
+    public static final String JOB_NAME = "CassandraMaintenanceJob";
+
+    public static final String CONFIG_ID = "org.hawkular.metrics.jobs." + JOB_NAME;
+
+    private ConfigurationService configurationService;
+
+    private EventDataAccess eventDataAccess;
+
+    private RxSession session;
+
+    public CassandraMaintenanceJob(RxSession session, EventDataAccess eventDataAccess,
+            ConfigurationService configurationService) {
+        this.session = session;
+        this.eventDataAccess = eventDataAccess;
+        this.configurationService = configurationService;
+    }
+
+    @Override
+    public Completable call(JobDetails jobDetails) {
+        logger.info("Starting Cassandra cluster maintenance");
+        Stopwatch stopwatch = Stopwatch.createStarted();
+
+        String jmxPort = jobDetails.getParameters().get("jmxPort");
+        String keyspace = jobDetails.getParameters().get("keyspace");
+
+        return eventDataAccess.findEventsForBucket(getCurrentBucket())
+                .collect(ClusterState::new, (state, event) -> {
+                    try {
+                        String hostname = event.getDetails().get("hostname");
+                        InetAddress address = InetAddress.getByName(hostname);
+                        state.setLastNodeAdded(address);
+                    } catch (UnknownHostException e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .flatMap(state -> {
+                    if (state.getLastNodeAdded() == null) {
+                        return Observable.empty();
+                    }
+                    return getNodes().filter(address -> !address.equals(state.getLastNodeAdded()));
+                })
+                .concatMap(address -> createExecutionContexts(address, getTables(keyspace), jmxPort, keyspace))
+                .flatMap(context -> new CleanupCommand().run(context))
+                .toCompletable()
+                .concatWith(Completable.defer(() -> configurationService.save(CONFIG_ID, "lastRun",
+                        Long.toString(jobDetails.getTrigger().getTriggerTime())).toCompletable()))
+                .doOnCompleted(() -> {
+                    stopwatch.stop();
+                    logger.infof("Finished Cassandra cluster maintenance in %d ms",
+                            stopwatch.elapsed(TimeUnit.MILLISECONDS));
+                })
+                .doOnError(t -> logger.warn("Cassandra cluster maintenance has been aborted to to an unexpected error",
```

Comment:
This initial commit is basically just a PoC. There will definitely be retry/failure logic.